From 7a5bbc08f44a226185ed9a53c3e57936bc78ebee Mon Sep 17 00:00:00 2001
From: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date: Tue, 11 Dec 2018 14:43:12 -0700
Subject: [PATCH] joyent/node-fast#22 Fix copy-paste bug in fast_server module
 Reviewed by: Kody A Kantor <kody@kkantor.com> Approved by: Kody A Kantor
 <kody@kkantor.com>

---
 CHANGES.md         | 4 ++++
 lib/fast_server.js | 2 +-
 package.json       | 2 +-
 test/tst.server.js | 3 ++-
 4 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 49f7d4d..2fbab3e 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,10 @@
 
 None yet.
 
+## v2.6.1
+
+* #22 Fix copy-paste bug in fast_server module
+
 ## v2.6.0
 
 * `fast_client_request_time_ms` became `fast_client_request_time_seconds` and is
diff --git a/lib/fast_server.js b/lib/fast_server.js
index 1c633cb..4cc50c0 100644
--- a/lib/fast_server.js
+++ b/lib/fast_server.js
@@ -227,7 +227,7 @@ function FastServer(args)
 	this.fs_nrequests_failed = 0;		/* count of reqs failed */
 
 	if (this.fs_collector) {
-		if (this.fc_collector.FIXED_BUCKETS === true) {
+		if (this.fs_collector.FIXED_BUCKETS === true) {
 			fixed_buckets = true;
 		}
 		this.fs_request_counter = this.fs_collector.counter({
diff --git a/package.json b/package.json
index c21d0f0..099d849 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "fast",
 	"description": "streaming JSON RPC over TCP",
-	"version": "2.6.0",
+	"version": "2.6.1",
 	"main": "./lib/fast.js",
 	"repository": {
 		"type": "git",
diff --git a/test/tst.server.js b/test/tst.server.js
index 67ee17e..964c314 100644
--- a/test/tst.server.js
+++ b/test/tst.server.js
@@ -168,6 +168,7 @@ function runTestCase(testcase, callback)
 	});
 	tctx.ts_socket = mod_net.createServer({ 'allowHalfOpen': true });
 	tctx.ts_server = new mod_fast.FastServer({
+            'collector': tctx.ts_collector,
 	    'log': tctx.ts_log.child({ 'component': 'FastServer' }),
 	    'server': tctx.ts_socket
 	});
@@ -1011,7 +1012,7 @@ serverTestCases = [ {
     'run': function (tctx, callback) {
 	function doBadRpc(client, cb) {
 		client.rpcBufferAndCallback({
-			'rpcmethod': '',
+			'rpcmethod': 'nonexistentrpcmethod',
 			'rpcargs': [],
 			'maxObjectsToBuffer': 0
 		}, function (err, data, ndata) {
-- 
2.21.0

